frontend/js/src/mbid-mapping/MbidMappingModal.tsx

```diff
@@ -0,0 +1,151 @@
+import * as React from "react";
+import { get as _get } from "lodash";
+import GlobalAppContext from "../utils/GlobalAppContext";
+import { getArtistName, getTrackName } from "../utils/utils";
+
+export type MbidMappingModalProps = {
+  recordingToMap?: Listen;
+  newAlert: (
+    alertType: AlertType,
+    title: string,
+    message: string | JSX.Element
+  ) => void;
+  onSuccessfulPin?: (pinnedrecording: PinnedRecording) => void;
+};
+
+export interface MbidMappingModalState {
+  recordingMBID: string;
+}
+
+export default class MbidMappingModal extends React.Component<
+  MbidMappingModalProps,
+  MbidMappingModalState
+> {
+  static contextType = GlobalAppContext;
+  declare context: React.ContextType<typeof GlobalAppContext>;
+
+  constructor(props: MbidMappingModalProps) {
+    super(props);
+    this.state = { recordingMBID: "" };
+  }
+
+  submitMbidMapping = async () => {
+    const { recordingToMap, newAlert, onSuccessfulPin } = this.props;
+    const { recordingMBID } = this.state;
+    const { APIService, currentUser } = this.context;
+
+    if (recordingToMap && currentUser?.auth_token) {
+      const recordingMSID = _get(
+        recordingToMap,
+        "track_metadata.additional_info.recording_msid"
+      );
+
+      try {
+        const response = await APIService.submitMbidMapping(
+          currentUser.auth_token,
+          recordingMSID,
+          recordingMBID
+        );
+        const { data } = response;
+      } catch (error) {
+        this.handleError(error, "Error while pinning track");
+        return;
+      }
+
+      newAlert(
+        "success",
+        `You mapped a track!`,
+        `${getArtistName(recordingToMap)} - ${getTrackName(recordingToMap)}`
+      );
+      this.setState({ recordingMBID: "" });
+
+      // if (onSuccessfulPin) {
+      //   onSuccessfulPin(newPin);
+      // }
+    }
+  };
+
+  handleMbidInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    const input = event.target.value;
```

Comment:
currently no validation on this input. We should do a regex to ensure that it's a valid uuid, and only allow submit to be set if so